vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java

```diff
@@ -1477,6 +1493,40 @@ public static String resolveHostNameInUrl(DatacenterMO dcMo, String url) {
         return url;
     }
 
+    public static String removeOVFNetwork(final String ovfString)  {
+        if (ovfString == null || ovfString.isEmpty()) {
+            return ovfString;
+        }
+        try {
+            final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            final Document doc = factory.newDocumentBuilder().parse(new ByteArrayInputStream(ovfString.getBytes()));
+            final DocumentTraversal traversal = (DocumentTraversal) doc;
+            final NodeIterator iterator = traversal.createNodeIterator(doc.getDocumentElement(), NodeFilter.SHOW_ELEMENT, null, true);
+            for (Node n = iterator.nextNode(); n != null; n = iterator.nextNode()) {
+                final Element e = (Element) n;
+                if ("NetworkSection".equals(e.getTagName())) {
+                    if (e.getParentNode() != null) {
+                        e.getParentNode().removeChild(e);
+                    }
+                } else if ("rasd:Connection".equals(e.getTagName())) {
+                    if (e.getParentNode() != null && e.getParentNode().getParentNode() != null) {
+                        e.getParentNode().getParentNode().removeChild(e.getParentNode());
+                    }
+                }
+            }
+            final DOMSource domSource = new DOMSource(doc);
+            final StringWriter writer = new StringWriter();
+            final StreamResult result = new StreamResult(writer);
+            final TransformerFactory tf = TransformerFactory.newInstance();
+            final Transformer transformer = tf.newTransformer();
+            transformer.transform(domSource, result);
+            return writer.toString();
+        } catch (SAXException | IOException | ParserConfigurationException | TransformerException e) {
+            s_logger.warn("Unexpected exception caught while removing network elements from OVF:", e);
```

Comment:
thanks for this.
I'd log a warn without, and a debug with the stacktrace.That way an admin had better control on what is seen in the logs.